dotnet/targets/Xamarin.Shared.Sdk.props

```diff
@@ -166,13 +166,14 @@
 
 	<PropertyGroup>
 		<!--
-			With NativeAOT we want to suppress trim warnings coming from ILLink and enable them only for ILC.
+			With NativeAOT we want to suppress trim warnings coming from ILLink and enable them only for ILC when publishing.
 			For this reason, in case of NativeAOT, we set SuppressTrimAnalysisWarnings to true by default and store the overwriten default in
 			_OriginalSuppressTrimAnalysisWarnings property, which is later used to properly configure warning suppression for ILC.
 		-->
 		<_OriginalSuppressTrimAnalysisWarnings>$(SuppressTrimAnalysisWarnings)</_OriginalSuppressTrimAnalysisWarnings>
-		<SuppressTrimAnalysisWarnings Condition="'$(_UseNativeAot)' == 'true'">true</SuppressTrimAnalysisWarnings>
-		<!-- Otherwise suppress trimmer warnings unless we're trimming all assemblies -->
+		<SuppressTrimAnalysisWarnings Condition="'$(PublishAot)' == 'true' and '$(_IsPublishing)' == 'true'">true</SuppressTrimAnalysisWarnings>
```

Comment:
Hm, but `_UseNativeAot=true` is set under the same condition:
https://github.com/xamarin/xamarin-macios/blob/af231730b4cf99ea699f2bb8d55a531bbe78d1bd/dotnet/targets/Xamarin.Shared.Sdk.props#L37

Reply:
I *think* what we would want is:

1. Remove the following line altogether: 

    ```xml
    <SuppressTrimAnalysisWarnings Condition="'$(_UseNativeAot)' == 'true'">true</SuppressTrimAnalysisWarnings>
    ```

2. Set `TrimMode=full` earlier in the SDK when `PublishAot` is enabled so that line 177 properly sets the value for `SuppressTrimAnalysisWarnings` and the analyzers are enabled for both build and publish targets

    ```xml
    <TrimMode Condition="'$(PublishAot)' == 'true'">full</TrimMode>
    ```


    NOTE: We are already setting the `TrimMode` below based on both `PublishAot` and `_IsPublishing` (i.e. through `_UseNativeAot`) but it is probably too late: https://github.com/xamarin/xamarin-macios/blob/af231730b4cf99ea699f2bb8d55a531bbe78d1bd/dotnet/targets/Xamarin.Shared.Sdk.props#L144 

3. Suppress warnings only for ILLinker when `_UseNativeAot=true` (ie. both `PublishAot=true` and `_IsPublishing=true`) in `_ComputeLinkerArguments` MSBuild target around:
https://github.com/xamarin/xamarin-macios/blob/af231730b4cf99ea699f2bb8d55a531bbe78d1bd/dotnet/targets/Xamarin.Shared.Sdk.targets#L530C8-L532

Reply:
> Hm, but _UseNativeAot=true is set under the same condition:

Yes, that's why there's the other condition:

```xml
<SuppressTrimAnalysisWarnings Condition="'$(SuppressTrimAnalysisWarnings)' == '' And '$(PublishAot)' == 'true'">false</SuppressTrimAnalysisWarnings>
```

I just wanted to expand `_UseNativeAot` in this case to avoid confusion why we're setting it to `true` in the `_UseNativeAot` case and then setting it to `false` in the `PublishAot` case.

> Set TrimMode=full when PublishAot=true so that line 177 properly sets the value for SuppressTrimAnalysisWarnings and the analyzers are enabled for both build and publish targets

Would there be any side effects of this change? As you pointed out, we currently aren't setting it in Debug builds. Could that slow down inner dev loop? Or is that irrelevant because we expect MAUI apps that use NativeAOT on iOS to also take advantage of full trimming on Android?